### ISSUE TITLE
NRE

### ISSUE THREAD

**Issue by NSTA1** [TNSGlobal/SPSS-.NET-Reader#10]
Got this NRE trying to instantiate SpssWriter if a Variable is missing the Name property.  (should probably throw an ArgumentException - took me a while to work out what was going on instead of the exception actually informing me what was going on)

Code:

```
            MemoryStream ms = new MemoryStream();

            var variables = new List<Variable>() {
                new Variable() {
                    Type = SpssLib.SpssDataset.DataType.Numeric,
                     Name = "Id"                      
                },
                new Variable() {
                    Type = SpssLib.SpssDataset.DataType.Text,
                    Width = 255,
                    Name = "Identifier"  // this was missing - NRE vanished as soon as supplied
                }
            };

            var options = new SpssOptions() { Compressed = true, DataEncoding = UTF8Encoding.UTF8, HeaderEncoding = UTF8Encoding.UTF8, Label = "Label", WeightVariable = variables[0] };

            using(var spss = new SpssWriter(ms, variables, options))   // NRE here
            {

            }
```

Stacktrace:

```
   at SpssLib.FileParser.Records.EncodingExtensions.AsCharArr(Encoding enc, String value, Int32 length, Int32& lastCharIndex, Int32& byteCount)
   at SpssLib.FileParser.Records.EncodingExtensions.AsCharArr(Encoding enc, String value, Int32 length, Int32& lastCharIndex)
   at SpssLib.FileParser.Records.EncodingExtensions.GetUpToMaxLenght(Encoding enc, String value, Int32 maxLength, Byte[] byteArr, Int32 byteIndex)
   at SpssLib.FileParser.Records.VariableDataInfoRecord`1.BuildDataArray()
   at SpssLib.FileParser.Records.VariableDataInfoRecord`1..ctor(IDictionary`2 dictionary, Encoding encoding)
   at SpssLib.FileParser.Records.LongVariableNamesRecord..ctor(IDictionary`2 variableLongNames, Encoding encoding)
   at SpssLib.FileParser.SavFileWriter.WriteFileHeader(SpssOptions options, IEnumerable`1 variables)
   at SpssLib.DataReader.SpssWriter.WriteFileHeader()
   at SpssLib.DataReader.SpssWriter..ctor(SavFileWriter output, ICollection`1 variables, SpssOptions options)
   at SpssLib.DataReader.SpssWriter..ctor(Stream output, ICollection`1 variables, SpssOptions options)
   at EUPillar.Model.NationalCaseRepository.ExportToSpss(SearchRequest sr) in c:\Users\lmi032\Documents\Visual Studio 2013\Projects\EUPillar\EUPillar.Model\NationalCase.cs:line 97
   at EUPillar.Controllers.ExportController.National(SearchRequest sr) in c:\Users\lmi032\Documents\Visual Studio 2013\Projects\EUPillar\EUPillar\Controllers\ExportController.cs:line 32
   at lambda_method(Closure , Object , Object[] )
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.<>c__DisplayClass10.<GetExecutor>b__9(Object instance, Object[] methodParameters)
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ActionExecutor.Execute(Object instance, Object[] arguments)
   at System.Web.Http.Controllers.ReflectedHttpActionDescriptor.ExecuteAsync(HttpControllerContext controllerContext, IDictionary`2 arguments, CancellationToken cancellationToken)
```

**Comment by ranieuwe** [comment]
This should indeed throw an ArgumentException because it is a required variable. Will fix!

**Comment by ranieuwe** [comment]
Thanks for the bug report. This has been fixed in version 1.1.1. NuGet package of it has been updated.
